src/librustc/mir/tcx.rs

```diff
@@ -195,7 +209,7 @@ impl<'tcx> Rvalue<'tcx> {
                     }
                 }
             }
-            &Rvalue::InlineAsm { .. } => None
+            Rvalue::InlineAsm { .. } => None
```

Comment:
>  Is there a strong reason for it not to be this way?

IIRC it has been this way all the way back to when first MIR patches landed. I have no idea why is it this way at the moment and I do not see any reason why it should stay an `Rvalue`.